### PR TITLE
Support return query result as a Json string

### PR DESCRIPTION
**Pull request by Aiee**
Design doc: https://confluence.nebula-graph.io/pages/viewpage.action?pageId=10725353

closing https://github.com/vesoft-inc/nebula/issues/2586

### CODE REVIEW

**Comment by Shylock-Hg** [comment]
Need end to end case in TCK.

**Comment by Aiee** [comment]
> Need end to end case in TCK.

Could you be more specific on this?

File: src/common/datatypes/Edge.cpp [hunk 1]
```diff
@@ -32,6 +32,45 @@ std::string Edge::toString() const {
...
+// Used in Json form query result
+// format:
+// {
+//   "id": {
+//      "name": _name,
+//      "src": srcVID,
+//      "dst": dstVID,
+//      "type": _type,
+//      "ranking": _rankding
+//    }
```
Comment (jievince) [inline src/common/datatypes/Edge.cpp:59]: "id" is used to be compatible with neo4j?

File: src/common/datatypes/DataSet.h [hunk 2]
```diff
@@ -153,6 +155,44 @@ struct DataSet {
...
+  // {
+  //   "row": [ row-data ],
+  //   "meta": [ metadata ]
+  // }
+  folly::dynamic rowToJsonObj(Row row) const {
+    folly::dynamic rowJsonObj = folly::dynamic::object();
+    auto rowDataList = folly::dynamic::array();
+    auto metaDataList = folly::dynamic::array();
+
+    for (const auto& ele : row.values) {
```
Comment (jievince) [inline src/common/datatypes/DataSet.h:186]: `ele` means `element`? A little hard to understand

File: src/common/datatypes/Edge.cpp (hunk 1 above)
Comment (Aiee) [reply src/common/datatypes/Edge.cpp:59]: Correct. In neo4j, `id` field for an edge is simply `"id": edgeID`.

File: src/common/datatypes/DataSet.h [hunk 3]
```diff
@@ -153,6 +155,44 @@ struct DataSet {
...
   }
 
+  // format:
+  // [
+  //   {
+  //     "row": [ row-data ],
+  //     "meta": [ metadata ]
+  //   },
+  // ]
+  folly::dynamic toJsonObj() const {
```
[the remark this answers is not in the document]
Comment (Aiee) [reply src/common/datatypes/DataSet.h:165]: fixed

**Review by yixinglu — APPROVED** [review]
LGTM

**Review by jievince — APPROVED** [review]
LGTM

**Comment by whitewum** [comment]
doc required?

**Comment by Aiee** [comment]
> doc required?

Not really here. Users are not able to trigger this functionality via sending queries.
I'll update the README of clients' repos, and this should be sufficient.

**Pull request merged**